src/gd.c

```diff
@@ -2339,8 +2342,8 @@ BGD_DECLARE(void) gdImageRectangle (gdImagePtr im, int x1, int y1, int x2, int y
 	}
 }
 
-BGD_DECLARE(void) gdImageFilledRectangle (gdImagePtr im, int x1, int y1, int x2, int y2,
-		int color)
+static void _gdImageFilledRectangle (gdImagePtr im, int x1, int y1, int x2, int y2,
+		int color, int horizontal)
```

Comment:
I would rename this function to

_gdFilledPatternRectangle to avoid confusion.

The last parameter should be called direction or axis to reflect what it actually does.

Only to know what it is in 3y from now. 😎